iac-extensions/arm/src/test/java/org/sonar/iac/arm/visitors/ArmSymbolVisitorTest.java

```diff
@@ -501,41 +646,38 @@ public ArmSourceCodeBuilder addVariableDeclaration(String variableDeclaration) {
     }
 
     public String build() {
-      return buildSourceCode(language, parameterDeclarations, variableDeclarations, outputValue);
-    }
-
-    private static String buildSourceCode(String language, List<String> parameterDeclarations, List<String> variableDeclarations,
-      @Nullable String outputValue) {
       if (JSON.equals(language)) {
-        return buildJsonFile(parameterDeclarations, variableDeclarations, outputValue);
+        return buildJsonFile();
       } else {
-        return buildBicepFile(parameterDeclarations, variableDeclarations, outputValue);
+        return buildBicepFile();
       }
     }
 
-    private static String buildBicepFile(List<String> parameterDeclarations, List<String> variableDeclarations,
-      @Nullable String outputValue) {
+    private String buildBicepFile() {
       String code = "";
-      if (!parameterDeclarations.isEmpty()) {
-        code = String.join("\n", parameterDeclarations);
-      }
-      if (!variableDeclarations.isEmpty()) {
+      code = addListOfStatements(code, parameterDeclarations);
+      code = addListOfStatements(code, variableDeclarations);
+      code = addListOfStatements(code, resourceDeclarations);
+      if (outputValue != null) {
         if (!code.isEmpty()) {
           code += "\n";
         }
-        code += String.join("\n", variableDeclarations);
+        code += outputValue;
       }
```

Comment:
It can be replaced by 
code = addListOfStatements(code, List.of(outputValue));

Reply:
Good catch! I need to wrap it into a null check, but it's cleaner :)